MIDI/talagan_MIDI CC Mapper X/MIDI CC Mapper X - Dump Current Function.lua

```diff
@@ -0,0 +1,192 @@
+-- @noindex
+-- @description Dumps the currently displayed curve from MIDI CC Mapper X.
+-- @about 
+-- @author Benjamin Talagan BAbut
+-- @version 1.0
+-- Licence: MIT
+-- REAPER: 5.0+
+-- Extensions: none
+
+local user_lib_path = "Data/talagan_MIDI CC Mapper X/func/user_lib"; 
+
+-- Split function
+local function csplit(str,sep)
+   local ret={}
+   local n=1
+   for w in str:gmatch("([^"..sep.."]*)") do
+      ret[n] = w
+      n = n + 1
+   end
+   return ret
+end
+
+local function getSaveParams(func)
+  func   = func or ""
+  
+  local retval, retval_csv = reaper.GetUserInputs(
+    "MIDI CC Mapper X : export...",
+    1, 
+    "Function sub path (without txt)", 
+    func);
+
+  local spath = csplit(retval_csv,'/');
+  func        = spath[#spath];
+  spath[#spath] = nil
+
+  return retval, table.concat(spath,"/"), func
+end
+
+local function file_exists(name)
+   local f=io.open(name,"r")
+   if f~=nil then io.close(f) return true else return false end
+end
+
+local function open_file_manager_on_path(path)
```

Comment:
Ouch, ok, that's a very good point, thanks a lot for having pinpointed it. I definitely need to correct this before release. The problem is that not everyone has SWS installed so I could check it first, with something like `reaper.NamedCommandLookup("_BR_VERSION_CHECK")` and if not, have a degraded behavior (a popup saying "you can open your resource directory" for example). What's your opinion, is it acceptable ?

Reply:
>  The problem is that not everyone has SWS installed so I could check it first, with something like

Yes, and also these functions are only available since SWS 2.9.8. This can be checked too as unknown API functions are nil since REAPER v5.30 (prior versions threw an error). (`APIExists` was added at the same time.)

Reply:
Thanks a lot for the advice! I used a combination of `APIExists` and `CF_LocateInExporer`, and implemented the mechanism we described above. That's definitely a nice enhancement :)